src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -5232,40 +5544,24 @@ private bool CheckTupleValEscape(ImmutableArray<BoundExpression> elements, uint
             return true;
         }
 
+#nullable enable
+
         private bool CheckValEscapeOfObjectInitializer(BoundObjectInitializerExpression initExpr, uint escapeFrom, uint escapeTo, BindingDiagnosticBag diagnostics)
         {
-            foreach (var expression in initExpr.Initializers)
+            foreach (var expr in initExpr.Initializers)
             {
-                if (expression.Kind == BoundKind.AssignmentOperator)
-                {
-                    var assignment = (BoundAssignmentOperator)expression;
-                    bool valid = assignment.IsRef
-                        ? CheckRefEscape(expression.Syntax, assignment.Right, escapeFrom, escapeTo, checkingReceiver: false, diagnostics: diagnostics)
-                        : CheckValEscape(expression.Syntax, assignment.Right, escapeFrom, escapeTo, checkingReceiver: false, diagnostics: diagnostics);
-
-                    if (!valid)
-                    {
-                        return false;
-                    }
-
-                    var left = (BoundObjectInitializerMember)assignment.Left;
-                    if (!CheckValEscape(left.Arguments, escapeFrom, escapeTo, diagnostics: diagnostics))
-                    {
-                        return false;
-                    }
-                }
-                else
+                if (GetValEscapeOfObjectMemberInitializer(expr, escapeFrom) > escapeTo)
                 {
-                    if (!CheckValEscape(expression.Syntax, expression, escapeFrom, escapeTo, checkingReceiver: false, diagnostics: diagnostics))
-                    {
-                        return false;
-                    }
+                    Error(diagnostics, _inUnsafeRegion ? ErrorCode.WRN_EscapeVariable : ErrorCode.ERR_EscapeVariable, initExpr.Syntax, expr.Syntax);
```

Comment:
This `Error` call deliberately has a different syntax than the previous implementation. The totality of the indexer can lead to the escape violation. I considered trying to replicate the old behavior by doing extra analysis here but it got decently complicated and the new error message feels more representative.